modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/provider/SystemRelatedModelsFDSDataProvider.java

```diff
@@ -90,28 +98,38 @@ public List<RelatedModel> getItems(
 				fdsPagination.getStartPosition(),
 				fdsPagination.getEndPosition()),
 			relatedModel -> {
-				String objectFieldDBColumnName =
-					objectDefinition.getPKObjectFieldDBColumnName();
+				ObjectField titleObjectField =
+					_objectFieldLocalService.fetchObjectField(
+						objectDefinition.getTitleObjectFieldId());
+
+				if (titleObjectField == null) {
+					titleObjectField =
+						_objectFieldLocalService.fetchObjectField(
+							objectDefinition.getObjectDefinitionId(), "id");
+				}
 
-				if (objectDefinition.getTitleObjectFieldId() > 0) {
-					ObjectField objectField =
-						_objectFieldLocalService.getObjectField(
-							objectDefinition.getTitleObjectFieldId());
+				User user = _userLocalService.getUser(
+					PrincipalThreadLocal.getUserId());
 
-					objectFieldDBColumnName = objectField.getDBColumnName();
-				}
+				Map<String, Object> values = ObjectEntryDTOConverterUtil.toDTO(
+					relatedModel, _dtoConverterRegistry,
+					objectDefinition.getName(),
+					_systemObjectDefinitionManagerRegistry, user);
 
-				Map<String, Object> modelAttributes =
-					relatedModel.getModelAttributes();
+				Object titleFieldValue =
+					ObjectEntryValuesUtil.getTitleFieldValue(
+						titleObjectField.getBusinessType(),
+						relatedModel.getModelAttributes(), titleObjectField,
+						user, values);
 
-				Object value = modelAttributes.get(objectFieldDBColumnName);
+				if (titleFieldValue == null) {
+					titleFieldValue = StringPool.BLANK;
+				}
 
 				return new RelatedModel(
 					objectDefinition.getClassName(),
-					GetterUtil.getLong(
-						modelAttributes.get(
-							objectDefinition.getPKObjectFieldDBColumnName())),
-					value.toString(), true);
+					GetterUtil.getLong(values.get("id")),
+					titleFieldValue.toString(), true);
```

Comment:
@devpedrotavares @seltong can we use `GetterUtil.getString()` instead?

Reply:
The titleFieldValue variable is an Object and it will not always be a an instance of a String. It could be a Boolean or Date, for example. If we call GetterUtil.getString() passing an Object that is not a String, it will return an empty string instead of calling toString() which is what we want.